cpu/stm32_common/periph/usbdev.c

```diff
@@ -0,0 +1,1151 @@
+            usbdev->usbdev.cb(&usbdev->usbdev, USBDEV_EVENT_SUSPEND);
+            usbdev->suspend = true;
+            _sleep_periph(conf);
+            DEBUG("PHY SUSP %lx\n", *_pcgcctl_reg(conf));
+            /* Disable USB clock */
+        }
+    }
+    else if (int_status & USB_OTG_GINTSTS_WKUINT) {
+        event = USB_OTG_GINTSTS_WKUINT;
+        if (usbdev->suspend) {
+            usbdev->suspend = false;
+            DEBUG("WAKE\n");
+            /* re-enable USB clock */
+            _wake_periph(conf);
+            usbdev->usbdev.cb(&usbdev->usbdev, USBDEV_EVENT_RESUME);
+        }
+    }
+
+    _global_regs(conf)->GINTSTS |= event;
+    _global_regs(conf)->GAHBCFG |= USB_OTG_GAHBCFG_GINT;
+}
+
+static void _usbdev_ep_init(usbdev_ep_t *ep)
+{
+    DEBUG("STM32 Initializing EP %u, %s\n", ep->num,
+          ep->dir == USB_EP_DIR_IN ? "IN" : "OUT");
+}
+
+static size_t _get_available(usbdev_ep_t *ep)
+{
+    stm32_usb_otg_fshs_t *usbdev = (stm32_usb_otg_fshs_t *)ep->dev;
+    const stm32_usb_otg_fshs_config_t *conf = usbdev->config;
+
+    return ep->len -
+           (_out_regs(conf, ep->num)->DOEPTSIZ & USB_OTG_DOEPTSIZ_XFRSIZ_Msk);
+}
+
+static int _usbdev_ep_get(usbdev_ep_t *ep, usbopt_ep_t opt,
+                          void *value, size_t max_len)
+{
+    (void)max_len;
+    int res = -ENOTSUP;
+    switch (opt) {
+        case USBOPT_EP_AVAILABLE:
+            assert(max_len == sizeof(size_t));
+            *(size_t *)value = _get_available(ep);
+            res = sizeof(size_t);
+            break;
+        default:
+            DEBUG("otg_fs: Unhandled get call: 0x%x\n", opt);
+            break;
+    }
+    return res;
+}
+
+static void _ep_set_stall(usbdev_ep_t *ep, bool enable)
+{
+    stm32_usb_otg_fshs_t *usbdev = (stm32_usb_otg_fshs_t *)ep->dev;
+    const stm32_usb_otg_fshs_config_t *conf = usbdev->config;
+
+    (void)enable;
+
+    if (ep->dir == USB_EP_DIR_IN) {
+        /* Disable first */
+        _ep_in_disable(conf, ep->num);
+        _in_regs(conf, ep->num)->DIEPCTL |= USB_OTG_DIEPCTL_STALL;
+    }
+    else {
+        /* Disable first */
+        _ep_out_disable(conf, ep->num);
+        _out_regs(conf, ep->num)->DOEPCTL |= USB_OTG_DOEPCTL_STALL;
+    }
+}
+
+static int _usbdev_ep_set(usbdev_ep_t *ep, usbopt_ep_t opt,
+                          const void *value, size_t value_len)
+{
+    (void)value_len;
+    int res = -ENOTSUP;
+    switch (opt) {
+        case USBOPT_EP_ENABLE:
+            assert(value_len == sizeof(usbopt_enable_t));
+            if (*((usbopt_enable_t *)value)) {
+                _ep_activate(ep);
+            }
+            else {
+                _ep_deactivate(ep);
+            }
+            res = sizeof(usbopt_enable_t);
+            break;
+        case USBOPT_EP_STALL:
+            assert(value_len == sizeof(usbopt_enable_t));
+            _ep_set_stall(ep, *(usbopt_enable_t *)value);
+            res = sizeof(usbopt_enable_t);
+            break;
+        default:
```

Comment:
Please add a DEBUG statement here.

Reply:
I've added the DEBUG statement and cleaned up the other debug statements where relevant.